File: csharp/Forca.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using UnityEngine.UI;

public class Forca : MonoBehaviour
{
   //array que receberá o conjunto de palavras
   public string [] palavra;

   //indice do array palavra 
   public int indice = 0;
   //palavra que foi sorteada aleatoriamente dentro do array palavra
   public string sorteada;
   //Objeto UI Text apontado na IDE para estas variáveis
   public Text Txt_Palavra;
   public Text Txt_Letra;
   //array que faz testes entre as opções do usuário e a palavra sorteada
   public string [] palavraTeste;
   //controle de tentaivas do usuário
   private int tentativas;

    void Start()
    {
       
        escolhePalavra();
        exibePalavra();

    }


    void escolhePalavra(){

        //atribui os valores à variável
        palavra  = new string[10]{"abelha", "calça", "cabrito", "calça", "ramo", "novilho", "caximbo", "prudente", "lanterna" , "gato"};
        
        //sorteio de um numero aleatório para o índice
        indice = Random.Range(0,10);
        //string que recebe a palavra sorteada
        sorteada = palavra[indice].ToString();

        //array redimensionado para o tamanho da palavra sorteada
        palavraTeste =  new string[sorteada.Length];
        //inicialização da variável
        tentativas = 0;


        for(int i=0; i<sorteada.Length; i++){

            //preenche o arrray com lacunas
            palavraTeste[i] = " __ ";

        }


    }


   public void verificaPalavra(){

        //controle do numero de tentativas conforme o quantitativo de letras  
        if(tentativas < sorteada.Length){  
          
                            //controle de ocorrências da letra inserida pelo usuário
                            int totalOcorrencias = 0;
                            //inicializa a string   
                            string aux = "";
                           
                            //Percorre o array para verificar a existência da letra inserida pelo usuário
                            for(int i=0; i<sorteada.Length; i++){

                                       //se existente a letra na palavra sorteada, armazena a letra e incrementa a tentativa
                                        if(sorteada.Substring(i,1).CompareTo(Txt_Letra.text) == 0 ){

                                            totalOcorrencias++;
                                            palavraTeste[i] = sorteada.Substring(i,1);
                                            aux = aux + " " + palavraTeste[i].ToString() +  " ";
                                        

                                        }

                                

                            }
                                       
                        
                            //Exibe a string montada pelo usuário no componente UI Text com lacunas
                            Txt_Palavra.text =  aux;
                            
                            //retira espaços e lacunas da variável para comparação
                            aux = aux.Replace(" ","");
                            aux = aux.Trim('_');         


                       
                      //compara a palavra montada pelo usuário com a sorteada
                       if(aux.CompareTo(sorteada)==0){

                            Txt_Palavra.text = "Você venceu:  " + aux;
                            
                       }
                       
                      
        }else{


            
               Txt_Palavra.text = "ERROU!! TENTE NOVAMENTE!";
              
            

        }
        
        


    }

    void exibePalavra(){

        int numLetras = sorteada.Length;
        string lacunas = "";
       

            for(int i=0; i<numLetras; i++){

                lacunas = lacunas + " ___ ";    
            
            }

        Txt_Palavra.text = lacunas;
           


    }


}

File: javascript/forca.js
/* 

FORCA JAVASCRIPT - RODAR NO BROWSER

Algoritmo simples em JavaScript ES6
que simula o jogo de forca. O objetivo
é descobrir as letras que compõe a palavra
escolhida aleatóriamente pelo código em uma
array. 

Objetivo: Testar o GIT quanto a implementações
de versões, compatilhamento de código e estudos.

by Cristiano Godoy
Nick: ChrisGodoy78
souzaegodoy@gmail.com

Data: 03/06/2020
*/

//Declaração de variáveis

let lacunas = ""        //representa a estrutura da palavra atraves de underscore
let op=0                //controla a duração do loop while
let guardaLetras = []   //recebe os acertos e lacunas vazias
let erros = 0           // controla a possibilidade de existencia de uma letra na palavra
let totalErros = 0      //controla o total de erros ao final de todas as tentativas

//armazenar um conjunto de palavras  em um array
const palavras = ['gato','toalha', 'armario', 'cama', 'bicicleta', 'banheiro', 'carro', 'escola', 'borrifador', 'cabelo']

//sortear uma palavra do array
let sorteio = palavras[Math.ceil(Math.random()* 11)]

//armazenar a quantidade de letras da palavra escolhida
let totalLetras = sorteio.length



            //Apresentar uma dica com a quantidade de letras em forma numérica
            document.write("\nDICA: Total de letras -> " + totalLetras)

                    //preenche a  string lacunas tantas vezes quanto o numero de 
                    //letras da palavra sorteada
                    for(i=0; i<totalLetras; i++){

                      lacunas = lacunas + " ____ "

                    }//final do for

                           //exib a string lacunas preenchida
                           document.write("\n<br>" + lacunas)
                           
                           
            





//Loop que realizara a entrada de dados do teclado com a letra digitada pelo usuário
for (a=1; a<=totalLetras; a++){

            //Atribui à variável local a letra digitada
            let letraUser = window.prompt("Palavra com "+ totalLetras +" letras! Digite uma letra: ")

              //exibe a letra escolhida              
              document.write("\n<br>Letra escolhida: " + letraUser)

                    //reseta as variáveis em cada passo do laço externo
                    lacunas = ""
                    erros = 0

                            //percorre todas as letras da palavra sorteada verificando sua existência
                            //e adicionando no array guardaLetras em sua devida posição
                            for (i=0; i<totalLetras; i++){
                                                
                                
                                
                                if(sorteio[i] === letraUser){
                                    
                                    //se encontra a ocorrência da letra guarda na respectiva posição
                                    //e exibe uma janela de dialogo do tipo alert parabenizando
                                    guardaLetras[i] = "_" + letraUser + "_"
                                     alert("Parabens! Voce acertou: " + guardaLetras)

                                }else{

                                    //se não encontrar a ocorrencia da letra, armazena no array
                                    //guardaLetras o underscore da lacuna na referida posição
                                    if(guardaLetras[i]){
                                            
                                            
                                            guardaLetras[i] = guardaLetras[i]
                                    }else{


                                            guardaLetras[i] = "___"

                                    }       
                                   
                                        //controla a inexistencia de ocorrência das letras
                                        erros++

                                                //verifica se há inexistência da ocorrencia da letra escolhida
                                                //na quantidade de letras da palavra sorteada, caso sejam iguais
                                                //contabiliza 1 erro
                                                if(erros == totalLetras){
                                                
                                                    totalErros++
                                                
                                                }
                        

                                }

                            }//final do for

          
        
    
    }//final do for externo
    



     //verifica se houveram erros
     if(totalErros == 0){

        //caso o usuario tenha acertado todas as letras exibe uma mensagem de parabens
        //com a palavra encontrada
       // alert("Parabéns!!! Você venceu!!! Palavra: " + guardaLetras)
        document.getElementById('ganhou').style.display='block'

     }else{   
        
       // alert("Infelizmente você não acertou!!! Tente novamente!")
        
        document.getElementById('perdeu').style.display='block'
        //exibe a palavra construida pelo usuário
        document.write("\n<br>Acertos: " + guardaLetras)
        //exibe o total de erros
        document.write("\n<br>Total de erros " + totalErros  )
        //exibe a imagem do boneco enforcado
        document.getElementById('imagem').style.display='block'
       
     }
     

     function mudaOpacidade(){

       

     }
     /*
     objetivos a alcançar:

     [ ] criar uma validacao para nao digitar a mesma letra
     [ ] implementar css e hml com a imagem da forca e o boneco conforme os erros
     [ ] implementar o javascript com elementos DOM para o uso de botões e input text no lugar de prompt
     [ ] implementar sons de erros e acertos

     */
       